### PR TITLE
feat(playwrighttesting): separate api, test data processing and utility layers

### PR DESCRIPTION
**Pull request by Sid200026**
Work Items

- Default run name is run id itself in case git-based information is not available
- Add API call retries and error handling
- Add tracing in API calls
- Scalable error handling and messaging
- Reporting error handling and messaging
- Refactor PlaywrightReporter file
- Improved logging

For easier review

- Scalable Error Handling - https://github.com/Azure/azure-sdk-for-net/commit/1d5a12fb7cda0b2252b10b30fd691fed68aa77e7
- API Retries and Tracing - https://github.com/Azure/azure-sdk-for-net/commit/f64347a1005d645eea45fce5ac04e2b95dddb86e
- Reporting API Failure Handling - https://github.com/Azure/azure-sdk-for-net/commit/ea1a3f1a1b85261e108e1894d96d97e4fb4988be

### CODE REVIEW

**Review by puagarwa — APPROVED** [review]
Looks Good.

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Implementation/ConsoleWriter.cs [hunk 1]
```diff
@@ -19,5 +19,17 @@ public void WriteLine(string? message = null)
                 Console.WriteLine(message);
             }
         }
+
+        public void WriteError(string? message = null)
```
Comment (kashish2508) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Implementation/ConsoleWriter.cs:23]: is if-else required here?

**Review by kashish2508 — APPROVED** [review]

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Implementation/Logger.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+internal class Logger : ILogger
+{
+    internal static bool EnableDebug { get { return !string.IsNullOrEmpty(Environment.GetEnvironmentVariable(Constants.s_pLAYWRIGHT_SERVICE_DEBUG)); } set { } }
+
+#pragma warning disable CA1822 // Mark members as static
+    private void Log(LogLevel level, string message)
+#pragma warning restore CA1822 // Mark members as static
+    {
+        if (EnableDebug)
```
[the remark this answers is not in the document]
Comment (Sid200026) [reply sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Implementation/Logger.cs]: We print error level logs for many cases which have no impact, so printing error logs by default would not be a good experience (We don't do it from js sdk as well).

Have added different logging levels - https://github.com/Azure/azure-sdk-for-net/pull/46681/commits/00e6f595fdf9dc8789376e21b3b0c12dbc28a4b0

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Constants.cs [hunk 3]
```diff
@@ -187,7 +187,7 @@ internal class Constants
     // Default constants
     internal static readonly string s_default_os = ServiceOs.Linux;
     internal static readonly string s_default_expose_network = "<loopback>";
-    internal static readonly string s_pLAYWRIGHT_SERVICE_DEBUG = "PLAYWRIGHT_SERVICE_DEBUG";
+    internal static readonly string s_pLAYWRIGHT_SERVICE_DEBUG = "Logging__LogLevel__MicrosoftPlaywrightTesting";
```
Comment (puagarwa) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Constants.cs:190]: Logging.LogLevel.MicrosoftPlaywrightTesting is this consistent with js package, ? if not then should we ? Does the OSS follow level pattern in thier DEBUG variable ? I wonder if we could follow similar pattern ?

Comment (Sid200026) [reply sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Constants.cs:190]: > is this consistent with js package

No, this is Azure SDK .NET standard

Comment (Sid200026) [reply sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Constants.cs:190]: https://learn.microsoft.com/en-us/dotnet/azure/sdk/logging

**Comment by Sid200026** [comment]
@microsoft-github-policy-service rerun

**Pull request merged**